packages/grafana-prometheus/src/components/monaco-query-field/monaco-completion-provider/situation.ts

```diff
@@ -560,10 +485,13 @@ export function getSituation(text: string, pos: number): Situation | null {
     // i do not use a foreach because i want to stop as soon
     // as i find something
     if (isPathMatch(resolver.path, ids)) {
-      // @ts-ignore
       return resolver.fun(currentNode, text, pos);
     }
   }
 
   return null;
 }
+
+function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+  return value !== null && value !== undefined;
+}
```

Comment:
It looks like we could lose the check for `undefined` here:

<img width="1078" alt="labelNodes notEmpty check has in input type Label or null" src="https://github.com/grafana/grafana/assets/5732000/2da61920-f66c-42d9-a41c-4ba98e2ee419">

```suggestion
function notEmpty<TValue>(value: TValue | null): value is TValue {
  return value !== null;
}
```

Reply:
You are right but I just want to be extra safe :) I think there is no harm to keep it like this way. WDYT?

Reply:
That's okay, there's no harm keeping it as-is :)

> I just want to be extra safe

To that point, it might be worth looking into more at some point...if `undefined` is possible here, we should make sure the types reflect that.